### ISSUE TITLE
USER ISSUE: Nested Clasuses in Contracts Crash

### ISSUE THREAD

**Issue by timzwiebel** [StrangeLoopGames/EcoIssues#24836]
\
**User Id: (Not logged user)**
**Version:** 0.10.1.5 beta release-619

**Steps to Reproduce:**
1. Create a contract with nested clauses and try to change the type of the nested clasuses

**Expected behavior:**
It should not crash

**Actual behavior:**
The following error message popped up:
```
Server encountered an exception:

<size=60.00%>Exception: ArgumentOutOfRangeException
Message:Specified argument was out of the range of valid values. (Parameter 'index')
Source:System.Collections.Immutable

System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'index')
   at System.Collections.Immutable.Requires.FailRange(String parameterName, String message)
   at System.Collections.Immutable.ImmutableList`1.Node.ItemRef(Int32 index)
   at Eco.Core.Utils.ThreadSafeList`1.get_Item(Int32 index)
   at Eco.Gameplay.Economy.Contracts.ClauseListExtensions.ChangeClause(IClauseList list, Int32 index, String newClauseType)
   at InvokeStub_ClauseListExtensions.ChangeClause(Object, Object, IntPtr*)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
O</size>...

Check log file to see full message.
```

**Do you have mods installed? Does the issue happen when no mods are installed?:**

**Please provide log files. Instruction is here https://github.com/StrangeLoopGames/EcoIssues/wiki/Getting-the-Eco-game-log-file

**Comment by Daniel-Magulado** [comment]
Hi @timzwiebel 

Thank you for submitting this bug report.
We're able to verify and confirm it.
Our development team is now looking into it and should roll out a fix in the coming game updates.

QA Ref - [ECO-33567](https://app.clickup.com/t/10591183/ECO-33567)

**Comment by D3nnis3n** [comment]
Fixed in Update 10.2.
